### DISCUSSION TITLE
So how do I use the soundcloud plugin?

### DISCUSSION THREAD

**Discussion by Galicarnax** [MusicPlayerDaemon/MPD#1700] [Q&A] [answered]
I have put this into `mpd.conf`:
```
playlist_plugin {
    name "soundcloud"
    enabled "yes"
    apikey "....."
}
```
And I checked that mpd is built with soundcloud plugin support (`soundcloud` is present under `Playlist plugins` section in the output of `mpd -V`.

Restarted `mpd`, and tried to load playlists from soundcloud with `mpc load http://soundcloud/...`, and also with (as per documentation) `mpc load soundcloud://...`. In all cases it says `MPD error: No such playlist`. Then I found it says the same if I put random string into `apikey`, or even if I remove the entire thing from the `mpd.conf`. So how do I even know if plugin is working?

**Comment by Galicarnax** [comment]
I've found a [post](https://www.runeaudio.com/forum/mpd-soundcloud-support-t944.html#p5617) from 9 years ago where someone succeded to use soundcloud plugin providing the uri in correct form. Still, that doesn't work for me, with the same `MPD error: No such playlist`. Trying to debug with `mpd -v --stderr --no-daemon`, I see the following:
```
client: process command "load "soundcloud://url/soundcloud.com/...<some_artist_page>...""
exception: Failed to parse JSON: lexical error: invalid char in json text.
exception: No such playlist
client: command returned 3
client: [3] process command list returned 3
client: [3] closed
```
(and, btw, they also write that omitting client_id/apikey has no effect).

So my guess is that by now this plugin is not maintained any more.

UPD: I have the same output quoted above if I completely remove the plugin's section from `mpd.conf`. Very confusing...

**Comment by MaxKellermann** [comment, marked as answer]
The only documentation that exists is in the commit message 10 years ago: https://github.com/MusicPlayerDaemon/MPD/commit/7cef52478dc2d5c5b9d2a75e34e4e57ca5cdb122
This was a one-time contribution by @robx and he never maintained it. Maybe the protocol has changed in the past 10 years, and maybe nobody but @robx has ever used it.

**Reply by Galicarnax** [reply]
Oh, thanks for clarification. Probably, then, it is a good idea to remove the relevant section from the documentation, or at least add a warning that plugin is unmaintained.

**Reply by robx** [reply]
Yeah, I'm not aware of anyone else using it. I did use it successfully for a while, but have not since many years.

Perhaps best remove the whole thing, rather than document it as unmaintained?

**Comment by perdakovich** [comment]
I would recommend Mopidy server to you, it has working SoundCloud extension and MPD support.

**Comment by skidoo23** [comment]
SoundCloud API did not change much. But they moved to OAuth2 authorization. Every request to SoundCloud API now  needs user token in request header.

Is there a (simple) way to make MPD send a custom http header from a playlist plugin?

**Comment by skidoo23** [comment]
Updated SoundCloud playlist. Please have a look at https://github.com/skidoo23/MPD/commits/soundcloud_oauth2/ Kindly asking for review and comments.

**Reply by kingosticks** [reply]
Are you planning to ask the Mopidy project if you can use their key (and quota)?

I spent a lot of time some years ago optimising Mopidy-Soundcloud to limit the API hits and stop our quota strangely running out, until I realised it was due to other people using our API key. Sadly, SoundCloud have done nothing to help us stop this happening, and here we are. I could waste more time making it harder for others to use our key but that's ultimately pointless.

What have you done to limit your quota usage to as little as possible?

**Comment by skidoo23** [comment]
I never intended to harm the Mopidy project. Therefore, I deleted the corresponding branch.

SoundCloud is currently not ready or able to issue new app IDs. On the other hand (according to the SoundCloud developer site) the only limit is 15,000 accesses to a track per 24 hours per app ID. On the other hand, virtually no one seemed to be using the SoundCloud plugin. Exceeding the limit therefore seemed very unlikely to me with a very small group of interested people.

**Comment by kingosticks** [comment]
Correct, they've been stuck in a useless limbo for literary years. However, if you contact them (via github) they claim to be providing new accounts in some situations. 

It is unlikely if you are careful about requests but I think there must be a load of people out there using it. Some of which may be using it stupidly and causing the quota problem. In any case, it's a mess.
